lib/flowViewport/sceneIndex/wireframeHighlights/fvpPiInstancerWhSi.cpp

```diff
@@ -443,9 +528,52 @@ void PiInstancerWhSi::_CreateSelectionHighlight(
     SdfPath selectionPath = RegisterSelection(selectionKey);
 
     SelectionData selectionData;
-    selectionData._primSelection = selectionId == kFullHighlight ? PrimSelection {instancerPath} : ConvertHydraToFvpSelection(instancerPath, selectionsSchema.GetElement(std::stoul(selectionId)));
     selectionData._instancerPaths = instancerPaths;
     selectionData._prototypePaths = prototypePaths;
+    
+    if (instanceMask.empty()) {
+        // Original behaviour: Handle different selection element access patterns and instance data
+        selectionData._primSelection = selectionId == kFullHighlight ? 
+            PrimSelection {instancerPath} : 
+            ConvertHydraToFvpSelection(instancerPath, selectionsSchema.GetElement(std::stoul(selectionId)));
+        selectionData._leadInstanceIndex = -1;
+        selectionData._activeInstanceIndices.clear();
+        
+        HdInstancerTopologySchema instancerTopology = HdInstancerTopologySchema::GetFromParent(instancerPrim.dataSource);
+        auto instanceIndices = instancerTopology.GetInstanceIndices();
+        size_t nbInstances = 0;
+        for (size_t iInstanceIndex = 0; iInstanceIndex < instanceIndices.GetNumElements(); iInstanceIndex++) {
+            auto protoInstances = instanceIndices.GetElement(iInstanceIndex)->GetTypedValue(0);
+            nbInstances += protoInstances.size();
+        }
```

Comment:
Can you factor this out into a function in the unnamed namespace, something like
```
size_t countNbInstances(const PXR_NS::HdSceneIndexPrim& instancerPrim)
```
And in the function documentation say that this is O(n) on the instance indices, since it loops over them?  Would make the code intent clearer.